File: 2018/02/1_1.rb
lines = File.readlines("input.txt")

twos = 0
threes = 0

lines.each do |line|
    counts = {}
    line.each_char do |char|
        counts[char] = (counts[char] or 0) + 1
    end
    twos += 1 if counts.value? 2
    threes += 1 if counts.value? 3
end

puts twos * threes

File: 2020/01/1_1.py
file = open('input.txt', 'r')
lines = file.readlines()

target = 2020
nums = {}

for line in lines:
    num = int(line)
    nums[num] = True

for num in nums:
    rem = target - num
    if rem > 0:
        if rem in nums:
            print(f"{num} {rem} = {num * rem}")
            break